client/components/ConfigurationDialog.jsx

```diff
@@ -0,0 +1,144 @@
+/**
+ * ConfigurationDialog dialog
+ */
+
+import React from 'react';
+import List from '@material-ui/core/List';
+import ListItem from '@material-ui/core/ListItem';
+import ListSubheader from '@material-ui/core/ListSubheader';
+import ListItemSecondaryAction from '@material-ui/core/ListItemSecondaryAction';
+import Switch from '@material-ui/core/Switch';
+import {ListItemText, Radio} from '../../node_modules/@material-ui/core';
+import Dialog from "@material-ui/core/Dialog";
+import DialogTitle from "@material-ui/core/DialogTitle";
+import DialogContent from "@material-ui/core/DialogContent";
+import FormControl from "@material-ui/core/FormControl";
+import FormLabel from "@material-ui/core/FormLabel";
+import RadioGroup from "@material-ui/core/RadioGroup";
+import DialogActions from "@material-ui/core/DialogActions";
+import FormControlLabel from "@material-ui/core/FormControlLabel";
+import Input from "@material-ui/core/Input";
+import Button from "@material-ui/core/Button";
+
+export default class ConfigurationDialog extends React.Component {
+
+  constructor(props, context) {
+    super(props, context);
+
+    this.state = {
+      disabledPipelines: new Set(this.props.disabledPipelines),
+      filterRegex: this.props.filterRegex,
+      pipelineNames: this.props.pipelineNames,
+      sortOrder: this.props.sortOrder,
+    };
+  }
+
+  isRegexValid() {
+    try {
+      new RegExp(this.state.filterRegex);
+      return true
+    } catch (_) {
+      return false
+    }
+  }
+
+  isPipelineSelectionDisabled(pipelineName) {
+    return !this.isRegexValid() || !pipelineName.match(this.state.filterRegex)
+  }
+
+  isPipelineToggledOn(pipelineName) {
+    return !this.isPipelineSelectionDisabled(pipelineName) && !this.state.disabledPipelines.has(pipelineName)
+  }
+
+  sortOrderChanged(e) {
+    this.setState({
+      sortOrder: e.target.value
+    })
+  }
+
+  regexChanged(e) {
+    this.setState({
+      filterRegex: e.target.value
+    })
+  }
+
+  togglePipeline(event) {
+    let pipelineName = event.target.value;
+    let pipelineEnabled = event.target.checked;
+    let newDisabledPipelines = new Set(this.state.disabledPipelines)
```

Comment:
IE doesn't supports `new Set(entries)` (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set#Browser_compatibility) and I don't have a polyfill. But on the other hand who uses IE nowadays? 🤷‍♂

Reply:
Ah, interesting :) Just pushed a commit to change it to plain old Array features. Not quite as elegant but still ok.